File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let min_num = prices[0]
    let res = 0
    for(let i=0; i<prices.length; i++){
        min_num = Math.min(min_num, prices[i])
        res = Math.max(res, prices[i] - min_num)
    }
    return res
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int res = 0, min = INT_MAX;
        for(int price: prices){
            if(price < min) min = price;
            else res = max(res, price - min);
        }
        return res;
    }
};